inc/saitunnel.h

```diff
@@ -335,7 +335,9 @@ typedef enum _sai_tunnel_ttl_mode_t
      *
      * Where the TTL field is preserved end-to-end by copying into the outer
      * header on encapsulation and copying from the outer header on
-     * decapsulation.
+     * decapsulation. This is applicable for inner IP packets. If the inner
+     * packet is a non-IP packet, then the value is undefined and implementation
+     * can chose a valid/meaningful outer TTL value, say in the case of VXLAN encap.
```

Comment:
Can't we define something, say on Switch/port level which is a default for all such cases? Or maybe follow the COS map rules as default behavior.

Reply:
@atit, this comment is solely to avoid the ambiguity. I don't think we would need to callout explicit default values in this particular case

Reply:
@atitjain, do you have some more concrete idea how to handle this case. we'd love to hear from you.

Reply:
Sure I will try.

Reply:
I think we need to update the behavioral model [Pipeline]((https://github.com/opencomputeproject/SAI/blob/master/doc/behavioral%20model/pipeline_v6.pdf)) with
1. CoS mapping at various stages (Port/ACL etc)
2. Pipeline Metadata part to reflect PCP/DSCP/TC/Color etc
Once this is done, this behavior will become deterministic.

Reply:
I think we need to update the [behavioral model](https://github.com/opencomputeproject/SAI/blob/master/doc/behavioral%20model/pipeline_v6.pdf) with
1. CoS mapping at various stages (Port/ACL etc)
2. Pipeline Metadata part to reflect PCP/DSCP/TC/Color etc
Once this is done, this behavior will become deterministic.

Reply:
@prsunny , can you update?